--- /usr/lib/python2.7/site-packages/cloudinit/config/cc_growpart.py
+++ cc_growpart.py
@@ -187,12 +187,14 @@
     rsyspath = os.path.realpath(syspath)
     disksyspath = os.path.dirname(rsyspath)
 
-    diskmajmin = util.load_file(os.path.join(disksyspath, "dev")).rstrip()
-    diskdevpath = os.path.realpath("/dev/block/%s" % diskmajmin)
+    return ("/dev/%s" % os.path.basename(disksyspath), ptnum)
 
+    #diskmajmin = util.load_file(os.path.join(disksyspath, "dev")).rstrip()
+    #diskdevpath = os.path.realpath("/dev/block/%s" % diskmajmin)
+
     # diskdevpath has something like 253:0
     # and udev has put links in /dev/block/253:0 to the device name in /dev/
-    return (diskdevpath, ptnum)
+    #return (diskdevpath, ptnum)
 
 
 def devent2dev(devent):
